src/Illuminate/Support/Collection.php

```diff
@@ -726,17 +726,19 @@ public function forPage($page, $perPage)
     }
 
     /**
-     * Partition the collection into two array using the given callback.
+     * Partition the collection into two arrays using the given callback or key.
      *
-     * @param  callable  $callback
-     * @return array
+     * @param  callable|string  $callback
+     * @return static
      */
-    public function partition(callable $callback)
+    public function partition($callback)
     {
-        $partitions = [new static(), new static()];
+        $partitions = new static([new static(), new static()]);
```

Comment:
1. We usually leave off the `()` for constructors with no arguments.
2. It'd be better to keep `$partitions` as a simple array, and only wrap it in a collection before returning. It's a minute performance improvement, but hey, why not.